tests/e2e/precompile-upgrade/suites.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// Implements precompile upgrade tests for Tx Allow List, requires network-runner cluster.
+package precompile_upgrade
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ava-labs/subnet-evm/precompile"
+	"github.com/ava-labs/subnet-evm/tests/e2e/utils"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+type chainConfig struct {
+	evm.Config
+}
+
+type chainUpgradeConfig struct {
+	// ref. https://docs.avax.network/subnets/subnet-upgrade#changing-subnet-configuration
+	PrecompileUpgrades []*params.PrecompileUpgrade `json:"precompileUpgrades,omitempty"`
+}
+
+var _ = utils.DescribeLocal("[Precompile Upgrade]", func() {
+	ginkgo.It("can upgrade for precompile", ginkgo.Label("precompile-upgrade"), func() {
+		runnerCli := utils.GetClient()
+		gomega.Expect(runnerCli).ShouldNot(gomega.BeNil())
+
+		defaultChainCfg := chainConfig{}
+		defaultChainCfg.Config.SetDefaults()
+		defaultChainCfg.Config.LogJSONFormat = true
+		defaultChainCfgBytes, err := json.Marshal(defaultChainCfg)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// "ewoq" key with "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+		ewoqPrivKey := "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
```

Comment:
nit: make this a constant

Reply:
Could also use the constant from AvalancheGo genesis directory and convert to the appropriate type here